src/physics/em/detail/LivermorePE.hh

```diff
@@ -26,12 +26,15 @@ namespace detail
 //---------------------------------------------------------------------------//
 /*!
  * Electron subshell data.
+ *
+ * The binding energy of consecutive shells is always decreasing.
```

Comment:
While this is typically true, there are a number of exceptions (at least one instance where this is false for ~40 different elements), so I don't think we can depend on it.

Reply:
Aha, thanks for that -- I thought based on your affirmation above that this was the case :( my restructuring of the sampling loop depended on that. (And this restructuring was precipitated by my working on a Selector class for abstracting the accumulate-select loop that shows up here and in several other classes.) I will try again...

Also it looks like H is an exception to the "xs below K shell" rule -- it only has xs_high -- that's what's failing the test now. I think rather than add a new "exception" to the rule in the runtime code, we should perhaps add a zero-XS value grid for hydrogen below that energy? Or is there a different behavior we need?

Reply:
Ahh I see now how that was misleading, sorry for the confusion :(.

Yeah, elements with only one subshell are missing the below-K-shell total cross section. At runtime it's only used in the `MicroXsCalculator`, and it should actually be ok if it's not present for those elements. If the incident photon energy is below the minimum binding energy, the minimum binding energy is used to calculate the micro xs instead -- this is done so that the cross section is constant rather than zero for low energy gammas, so they can always be "seen". Since H _only_ has the K shell, the energy used to calculate the micro xs will always be >= the K shell binding energy, and `xs_lo` will never be needed.

Reply:
OK perfect, glad you thought through all this! 😄 I missed the check for being below the minimum binding energy, which makes sense.

Reply:
Clearly I should have documented some of this better, so I'm glad you're going through it 😆

Reply:
Clearly I should have reviewed it better and asked these questions then! 😅 This has been a complete 🐰 🕳️  on my part so thanks for bearing with me... there really was nothing at all wrong with the code as it was, but I wanted to try to change it to use this new Selector class... and might end up not using it at all, at the end of the day 😂 but at least we'll have some more documentation.